torch/utils/cpp_extension.py

```diff
@@ -2422,3 +2481,518 @@ def _is_cuda_file(path: str) -> bool:
+
+    def check_onemkl_cfg(self):
+        if not self.__onemkl_root:
+            raise RuntimeError("Didn't detect mkl root. Please source <oneapi_dir>/mkl/<version>/env/vars.sh ")
+
+    def check_onednn_cfg(self):
+        if not self.__onednn_root:
+            raise RuntimeError("Didn't detect dnnl root. Please source <oneapi_dir>/dnnl/<version>/env/vars.sh ")
+        else:
+            warnings.warn(
+                "This extension has static linked onednn library. Please attaction to \
+                that, this path of onednn version maybe not match with the built-in version."
+            )
+
+    def check_sycl_cfg(self):
+        if not self.__sycl_root:
+            raise RuntimeError("Didn't detect sycl root. Please source <oneapi_dir>/compiler/<version>/env/vars.sh ")
+
+    def get_default_include_dir(self) -> List[str]:
+        return [os.path.join(self.__default_root, "include")]
+
+    def get_default_lib_dir(self) -> List[str]:
+        return [os.path.join(self.__default_root, "lib")]
+
+    def get_sycl_include_dir(self) -> List[str]:
+        return [
+            os.path.join(self.__sycl_root, "linux", "include"),
+            os.path.join(self.__sycl_root, "linux", "include", "sycl"),
+        ]
+
+    def get_onemkl_include_dir(self) -> List[str]:
+        return [os.path.join(self.__onemkl_root, "include")]
+
+    def get_onednn_include_dir(self) -> List[str]:
+        return [os.path.join(self.__onednn_root, "include")]
+
+    def get_onednn_lib_dir(self) -> List[str]:
+        return [os.path.join(self.__onednn_root, "lib")]
+
+    def is_onemkl_ready(self):
+        if self.__onemkl_root is None:
+            return False
+        return True
+
+    def is_onednn_ready(self):
+        if self.__onednn_root is None:
+            return False
+        return True
+
+    def get_library_dirs(self):
+        library_dirs = []
+        library_dirs += [f"{x}" for x in self.get_default_lib_dir()]
+        library_dirs += [f"{x}" for x in self.get_onednn_lib_dir()]
+        return library_dirs
+
+    def get_include_dirs(self):
+        include_dirs = []
+        include_dirs += [f"{x}" for x in self.get_sycl_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_onemkl_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_onednn_include_dir()]
+        include_dirs += [f"{x}" for x in self.get_default_include_dir()]
+        return include_dirs
+
+    def get_onemkl_libraries(self):
+        MKLROOT = self.__onemkl_root
+        return [
+            f"{MKLROOT}/lib/intel64/libmkl_sycl.a",
+            f"{MKLROOT}/lib/intel64/libmkl_intel_ilp64.a",
+            f"{MKLROOT}/lib/intel64/libmkl_sequential.a",
+            f"{MKLROOT}/lib/intel64/libmkl_core.a",
+        ]
+
+
+def _get_one_api_help():
+    oneAPI = _one_api_help()
+    return oneAPI
+
+def XPUExtension(name, sources, *args, **kwargs):
+    r"""
+    Creates a :class:`setuptools.Extension` for XPU/C++.
```

Comment:
It's unclear what's the primary purpose of this extension class is. There is no such programming language as XPU. So, documentation of this extension should talk about supporting SYCL kernels at the first place. Further this extension as currently written enables support for few other things as well such as oneDNN. This also should be called out in documentation explicitly.